src/dcfldd_error.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * dcfldd_error.c -- replacement for non-portable GNU libc's error()
+ * Copyright (C) 2024 David da Silva Polverari
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+#include "config.h"
+
+#include <assert.h>
+#include <errno.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "dcfldd.h"	// to use program_name
```

Comment:
(could be moved up right after `config.h` but just an idea)

Reply:
Hmm, I usually like to include user include files after system ones, although `config.h` is a special case, as `assert.h` will need to know whether `NDEBUG` is defined before being included. That's why I included `config.h` before the system includes, and `dcfldd.h` after. Does that feel right?

Reply:
@davidpolverari nevermind, it's intentional then, alright :+1: